swarm/api/manifest.go

```diff
@@ -62,6 +62,11 @@ func readManifest(manifestReader storage.LazySectionReader, hash storage.Key, dp
 
 	// TODO check size for oversized manifests
 	size, err := manifestReader.Size(quitC)
+	if err != nil { // size == 0
+		// can't determine size means we don't have the root chunk
+		glog.V(logger.Detail).Infof("Manifest %v not found.", hash.Log())
+		return
```

Comment:
Would it be better to return a descriptive error here?

There is also no need to log the error. In Go, *handling* errors (i.e. by logging them) should usually be deferred to the place where the error ends up because more context is available. `readManifest` might also be called in a setting where errors don't matter. In that case the logging does more harm than good.